It's a combination of errors about non-existing methods used in this library itself and errors from dependencies of `nalgebra`.

<details>
<summary>Complete error output</summary>

```
error[E0463]: can't find crate for `std`
  |
  = note: the `thumbv6m-none-eabi` target may not support the standard library
  = note: `std` is required by `matrixmultiply` because it does not declare `#![no_std]`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`

error[E0463]: can't find crate for `std`
 --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/aligned_alloc.rs:5:5
  |
5 | use std::alloc::{self, handle_alloc_error, Layout};
  |     ^^^ can't find crate
  |
  = note: the `thumbv6m-none-eabi` target may not support the standard library
  = help: consider building the standard library from source with `cargo build -Zbuild-std`

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:235:5
    |
235 |     assert!(mr > 0 && mr <= 8);
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:236:5
    |
236 |     assert!(nr > 0 && nr <= 8);
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:237:5
    |
237 |     assert!(mr * nr * size_of::<K::Elem>() <= 8 * 4 * 8);
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:238:5
    |
238 |     assert!(K::align_to() <= 32);
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:241:5
    |
241 |     assert!(K::align_to() <= max_align);
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:243:5
    |
243 |     assert!(K::MR <= K::mc());
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:244:5
    |
244 |     assert!(K::mc() <= K::kc());
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:245:5
    |
245 |     assert!(K::kc() <= K::nc());
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:246:5
    |
246 |     assert!(K::nc() <= 65536);
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `debug_assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:263:5
    |
263 |     debug_assert!(m <= 1 || n == 0 || rsc != 0);
    |     ^^^^^^^^^^^^
    |
help: consider importing this macro
    |
9   + use core::debug_assert;
    |

error: cannot find macro `debug_assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:264:5
    |
264 |     debug_assert!(m == 0 || n <= 1 || csc != 0);
    |     ^^^^^^^^^^^^
    |
help: consider importing this macro
    |
9   + use core::debug_assert;
    |

error: cannot find macro `debug_assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:314:21
    |
314 |                     debug_assert!(i < nap);
    |                     ^^^^^^^^^^^^
    |
help: consider importing this macro
    |
9   + use core::debug_assert;
    |

error: cannot find macro `thread_local` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:355:1
    |
355 | thread_local! {
    | ^^^^^^^^^^^^
    |
    = note: `thread_local` is in scope, but it is an attribute: `#[thread_local]`

error: cannot find macro `assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:378:5
    |
378 |     assert!(mr * nr * size_of::<K::Elem>() <= KERNEL_MAX_SIZE && K::align_to() <= KERNEL_MAX_ALIGN);
    |     ^^^^^^
    |
help: consider importing this macro
    |
9   + use core::assert;
    |

error: cannot find macro `debug_assert_ne` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:445:5
    |
445 |     debug_assert_ne!(na, 0);
    |     ^^^^^^^^^^^^^^^
    |
help: consider importing this macro
    |
9   + use core::debug_assert_ne;
    |

error: cannot find macro `debug_assert` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:446:5
    |
446 |     debug_assert!(na <= 128);
    |     ^^^^^^^^^^^^
    |
help: consider importing this macro
    |
9   + use core::debug_assert;
    |

error: cannot find macro `cfg` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:464:8
    |
464 |     if cfg!(target_os = "macos") {
    |        ^^^
    |
    = note: `cfg` is in scope, but it is an attribute: `#[cfg]`
help: consider importing this macro
    |
9   + use core::cfg;
    |

error[E0463]: can't find crate for `std`
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/lib.rs:30:1
   |
30 | extern crate std;
   | ^^^^^^^^^^^^^^^^^ can't find crate
   |
   = note: the `thumbv6m-none-eabi` target may not support the standard library
   = help: consider building the standard library from source with `cargo build -Zbuild-std`

error: cannot find attribute `derive` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/ptr.rs:14:3
   |
14 | #[derive(Copy, Clone)]
   |   ^^^^^^
   |
help: consider importing this attribute macro
   |
11 + use core::prelude::rust_2024::derive;
   |

error: cannot find attribute `derive` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/threading.rs:92:3
   |
92 | #[derive(Copy, Clone)]
   |   ^^^^^^
   |
help: consider importing this attribute macro
   |
16 + use core::prelude::rust_2024::derive;
   |

error: cannot find attribute `derive` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/util.rs:11:3
   |
11 | #[derive(Copy, Clone)]
   |   ^^^^^^
   |
help: consider importing this attribute macro
   |
9  + use core::prelude::rust_2024::derive;
   |

error: cannot find macro `debug_assert_eq` in this scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/dgemm_kernel.rs:1023:5
     |
1023 |     debug_assert_eq!(beta, 0., "Beta must be 0 or is not masked");
     |     ^^^^^^^^^^^^^^^
     |
help: consider importing this macro
     |
9    + use core::debug_assert_eq;
     |

error: cannot find macro `debug_assert_eq` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/sgemm_kernel.rs:708:5
    |
708 |     debug_assert_eq!(beta, 0., "Beta must be 0 or is not masked");
    |     ^^^^^^^^^^^^^^^
    |
help: consider importing this macro
    |
9   + use core::debug_assert_eq;
    |

error[E0425]: cannot find value `MASK_BUF` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:395:23
    |
395 |                 ptr = MASK_BUF.with(|buf| (*buf.get()).buffer.as_mut_ptr());
    |                       ^^^^^^^^ not found in this scope

error[E0433]: failed to resolve: use of undeclared type `Ord`
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/gemm.rs:465:20
    |
465 |         align_to = Ord::max(align_to, 8);
    |                    ^^^ use of undeclared type `Ord`
    |
help: consider importing this trait
    |
9   + use core::cmp::Ord;
    |

error[E0405]: cannot find trait `Copy` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/kernel.rs:94:28
   |
94 | pub(crate) trait Element : Copy + Send + Sync {
   |                            ^^^^ not found in this scope
   |
help: consider importing this trait
   |
9  + use core::marker::Copy;
   |

error[E0405]: cannot find trait `Send` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/kernel.rs:94:35
   |
94 | pub(crate) trait Element : Copy + Send + Sync {
   |                                   ^^^^ not found in this scope
   |
help: consider importing this trait
   |
9  + use core::marker::Send;
   |

error[E0405]: cannot find trait `Sync` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/kernel.rs:94:42
   |
94 | pub(crate) trait Element : Copy + Send + Sync {
   |                                          ^^^^ not found in this scope
   |
help: consider importing this trait
   |
9  + use core::marker::Sync;
   |

error[E0405]: cannot find trait `Sync` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/ptr.rs:17:16
   |
17 | unsafe impl<T> Sync for Ptr<*const T> { }
   |                ^^^^ not found in this scope
   |
help: consider importing this trait
   |
11 + use core::marker::Sync;
   |

error[E0405]: cannot find trait `Sync` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/ptr.rs:18:16
   |
18 | unsafe impl<T> Sync for Ptr<*mut T> { }
   |                ^^^^ not found in this scope
   |
help: consider importing this trait
   |
11 + use core::marker::Sync;
   |

error[E0405]: cannot find trait `Send` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/ptr.rs:19:16
   |
19 | unsafe impl<T> Send for Ptr<*const T> { }
   |                ^^^^ not found in this scope
   |
help: consider importing this trait
   |
11 + use core::marker::Send;
   |

error[E0405]: cannot find trait `Send` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/ptr.rs:20:16
   |
20 | unsafe impl<T> Send for Ptr<*mut T> { }
   |                ^^^^ not found in this scope
   |
help: consider importing this trait
   |
11 + use core::marker::Send;
   |

error[E0405]: cannot find trait `Copy` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/ptr.rs:33:18
   |
33 |         where T: Copy
   |                  ^^^^ not found in this scope
   |
help: consider importing this trait
   |
11 + use core::marker::Copy;
   |

error[E0405]: cannot find trait `FnOnce` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/threading.rs:202:19
    |
202 |         where G2: FnOnce(usize, usize) -> R + Sync
    |                   ^^^^^^ not found in this scope
    |
help: consider importing this trait
    |
16  + use core::ops::FnOnce;
    |

error[E0405]: cannot find trait `Sync` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/threading.rs:202:47
    |
202 |         where G2: FnOnce(usize, usize) -> R + Sync
    |                                               ^^^^ not found in this scope
    |
help: consider importing this trait
    |
16  + use core::marker::Sync;
    |

error[E0405]: cannot find trait `FnOnce` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/threading.rs:215:14
    |
215 |     where G: FnOnce(usize, usize) -> R + Sync,
    |              ^^^^^^ not found in this scope
    |
help: consider importing this trait
    |
16  + use core::ops::FnOnce;
    |

error[E0405]: cannot find trait `Sync` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/threading.rs:215:42
    |
215 |     where G: FnOnce(usize, usize) -> R + Sync,
    |                                          ^^^^ not found in this scope
    |
help: consider importing this trait
    |
16  + use core::marker::Sync;
    |

error[E0405]: cannot find trait `Fn` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/threading.rs:218:18
    |
218 |         where F: Fn(ThreadPoolCtx<'_>, &mut R, usize, usize) + Sync,
    |                  ^^ not found in this scope
    |
help: consider importing this trait
    |
16  + use core::ops::Fn;
    |

error[E0405]: cannot find trait `Sync` in this scope
   --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/threading.rs:218:64
    |
218 |         where F: Fn(ThreadPoolCtx<'_>, &mut R, usize, usize) + Sync,
    |                                                                ^^^^ not found in this scope
    |
help: consider importing this trait
    |
16  + use core::marker::Sync;
    |

error[E0405]: cannot find trait `Drop` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/aligned_alloc.rs:55:9
   |
55 | impl<T> Drop for Alloc<T> {
   |         ^^^^ not found in this scope
   |
help: consider importing this trait
   |
3  + use core::ops::Drop;
   |

error[E0405]: cannot find trait `Iterator` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/util.rs:24:6
   |
24 | impl Iterator for RangeChunk {
   |      ^^^^^^^^ not found in this scope
   |
help: consider importing this trait
   |
9  + use core::iter::Iterator;
   |

error[E0412]: cannot find type `Option` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/util.rs:28:27
   |
28 |     fn next(&mut self) -> Option<Self::Item> {
   |                           ^^^^^^ not found in this scope
   |
help: consider importing this enum
   |
9  + use core::option::Option;
   |

error[E0425]: cannot find value `None` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/util.rs:30:13
   |
30 |             None
   |             ^^^^ not found in this scope
   |
help: consider importing this unit variant
   |
9  + use core::option::Option::None;
   |

error[E0425]: cannot find function, tuple struct or tuple variant `Some` in this scope
  --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/matrixmultiply-0.3.8/src/util.rs:36:13
   |
36 |             Some((i, rem))
   |             ^^^^ not found in this scope
   |
help: consider importing this tuple variant
   |
9  + use core::option::Option::Some;
   |

error: aborting due to 45 previous errors

Some errors have detailed explanations: E0405, E0412, E0425, E0433, E0463.
For more information about an error, try `rustc --explain E0405`.
error[E0599]: no method named `sqrt` found for type `f32` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f32x4_.rs:1197:29
     |
1197 |           1.0 / self.arr[0].sqrt(),
     |                             ^^^^ method not found in `f32`

error[E0599]: no method named `sqrt` found for type `f32` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f32x4_.rs:1198:29
     |
1198 |           1.0 / self.arr[1].sqrt(),
     |                             ^^^^ method not found in `f32`

error[E0599]: no method named `sqrt` found for type `f32` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f32x4_.rs:1199:29
     |
1199 |           1.0 / self.arr[2].sqrt(),
     |                             ^^^^ method not found in `f32`

error[E0599]: no method named `sqrt` found for type `f32` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f32x4_.rs:1200:29
     |
1200 |           1.0 / self.arr[3].sqrt(),
     |                             ^^^^ method not found in `f32`

error[E0599]: no method named `sqrt` found for type `f32` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f32x4_.rs:1224:23
     |
1224 |           self.arr[0].sqrt(),
     |                       ^^^^ method not found in `f32`

error[E0599]: no method named `sqrt` found for type `f32` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f32x4_.rs:1225:23
     |
1225 |           self.arr[1].sqrt(),
     |                       ^^^^ method not found in `f32`

error[E0599]: no method named `sqrt` found for type `f32` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f32x4_.rs:1226:23
     |
1226 |           self.arr[2].sqrt(),
     |                       ^^^^ method not found in `f32`

error[E0599]: no method named `sqrt` found for type `f32` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f32x4_.rs:1227:23
     |
1227 |           self.arr[3].sqrt(),
     |                       ^^^^ method not found in `f32`

error[E0599]: no method named `sqrt` found for type `f64` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f64x2_.rs:1236:23
     |
1236 |           self.arr[0].sqrt(),
     |                       ^^^^ method not found in `f64`

error[E0599]: no method named `sqrt` found for type `f64` in the current scope
    --> /home/alexander/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wide-0.7.13/src/f64x2_.rs:1237:23
     |
1237 |           self.arr[1].sqrt(),
     |                       ^^^^ method not found in `f64`

error: aborting due to 11 previous errors

Some errors have detailed explanations: E0463, E0599.
For more information about an error, try `rustc --explain E0463`.
```
</details>

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -14,10 +14,10 @@ jobs:
       matrix:
         rust: [stable]
         os: [ubuntu-latest, macOS-latest, windows-latest]
-        features: [--features default, --no-default-features]
+        features: [--features default, --no-default-features --feature no_std]
         include:
           - {rust: nightly, os: ubuntu-latest, features: --features default}
-          - {rust: nightly, os: ubuntu-latest, features: --no-default-features}
+          - {rust: nightly, os: ubuntu-latest, features: --no-default-features --feature no_std}
     steps:
       - uses: actions/checkout@v2
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -14,10 +14,10 @@ jobs:
       matrix:
         rust: [stable]
         os: [ubuntu-latest, macOS-latest, windows-latest]
-        features: [--features default, --no-default-features --feature no_std]
+        features: [--features default, --no-default-features --features no_std]
         include:
           - {rust: nightly, os: ubuntu-latest, features: --features default}
-          - {rust: nightly, os: ubuntu-latest, features: --no-default-features --feature no_std}
+          - {rust: nightly, os: ubuntu-latest, features: --no-default-features --features no_std}
     steps:
       - uses: actions/checkout@v2
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "eskf"
-version = "0.2.0"
+version = "0.3.0"
 authors = ["Jørgen Nordmoen <jorgen@nordmoen.net>"]
 description = "Navigation filter based on an Error State Kalman Filter (ESKF)"
 keywords = ["navigation", "filter", "orientation", "kalman"]

